packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -16212,6 +16212,35 @@ void main() {
       skip: !kIsWeb, // [intended]
     );
 
+    // Regression test for https://github.com/flutter/flutter/issues/163399.
+    testWidgets('when selectAllOnFocus is turned off', (WidgetTester tester) async {
```

Comment:
Can you also test when this is set to true? With my comment about how the parameter should apply equally to all platforms, the behavior of `true` is different than the default behavior.